src/client.cpp

```diff
@@ -260,6 +260,7 @@ void Client::connect(Address address,
 	DSTACK(FUNCTION_NAME);
 
 	initLocalMapSaving(address, address_name, is_local_server);
+	m_address_name = address_name;
```

Comment:
because you stored that information, why not setting this member earlier and remove param from connect function ?